### PR TITLE
Fix 989 implement PreFilterAttributes and PreFilterEvents events

### LINKED ISSUE #989: [.Net Core] ComponentDesigner is not fully ported
* .NET Core Version: 3.0.0-preview6-27712-03
* Have you experienced this same bug with .NET Framework?: No

**Problem description:**
ComponentDesigner currently has not implemented methods PreFilterAttributes and PreFilterEvents.
https://github.com/dotnet/winforms/blob/master/src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ComponentDesigner.cs#L225

And I have a question about IDesignerFilter methods implementation in ComponentDesigner. Shouldn't it call the corresponding virtual methods? (the way it's done in the .NET Framework)

**Actual behavior:**
Throws NotImplementedException

**Expected behavior:**
Implemented :)

### PR DESCRIPTION
**Pull request by RussKie**
Closes #989

## Proposed changes

- Copy the behaviour as provided by NET Fx.
- Cleanup remove duplicated code and invert `if`s to reduce nesting

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ComponentDesigner.cs [hunk 1]
```diff
@@ -157,110 +157,32 @@ public virtual void InitializeNewComponent(IDictionary defaultValues)
...
-                for (int i = 0; i < values.Length; i++)
-                {
-                    EventDescriptor evt = values[i];
-                    if (evt != null)
-                    {
-                        events[evt.Name] = TypeDescriptor.CreateEvent(evt.ComponentType, evt, ReadOnlyAttribute.Yes);
-                    }
-                }
-            }
+            PostFilterEvents(events);
```
Comment (hughbe) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ComponentDesigner.cs]: Isn't this potentially breaking. Where we didn't call the virtual method we now do?

Comment (weltkante) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ComponentDesigner.cs]: If I'm not mistaken Desktop Framework also calls the virtual method

Comment (RussKie) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ComponentDesigner.cs]: The changes are a direct port from what we have in NET Fx codebase, so if anything we now made it "correct".

Comment (dreddy-work) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ComponentDesigner.cs]: This is what i was looking on Framework code. I am not sure why this was changed on the initial port.  @Tanya-Solyanik

Comment (RussKie) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ComponentDesigner.cs]: http://index/?query=PostFilterEvents&rightProject=System.Design&file=ComponentDesigner.cs&rightSymbol=ynjwyyat84aq

Comment (Tanya-Solyanik) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ComponentDesigner.cs]: >I am not sure why this was changed on the initial port. 

In general initial port included only APIs that had confirmed usage in Core. These were not in the list and thus were not ported. I can't comment on this particular implementation.

**Pull request merged**